amazon/ion/core.py

```diff
@@ -154,6 +154,25 @@ def derive_annotations(self, annotations):
             self.depth
         )
 
+    def derive_value(self, value):
+        """Derives a new event from this one setting the ``value`` attribute.
+
+        Args:
+            value: (any):
+                The value associated with the derived event.
+
+        Returns:
+            IonEvent: The newly generated event.
+        """
+        return IonEvent(
```

Comment:
Given `IonThunkEvent` we probably should be explicit that this creates a non-thunk event.